### ISSUE TITLE
Improve test coverage of sink values

### ISSUE THREAD

**Issue by lukewarlow** [w3c/trusted-types#494]
We should ensure that we have exhaustive coverage of the "sink" value, this is the prefix for violation object samples, aswell as being one of the arguments for the default policy.

**Comment by lukewarlow** [comment]
Example change https://github.com/web-platform-tests/wpt/pull/45058

**Comment by fred-wang** [comment]
Trusted Types spec:
https://w3c.github.io/trusted-types/dist/spec/#enforcement-in-scripts

HTML spec:
https://html.spec.whatwg.org/multipage/infrastructure.html#tt-trustedhtml
https://html.spec.whatwg.org/multipage/infrastructure.html#tt-trustedscript
https://html.spec.whatwg.org/multipage/infrastructure.html#tt-trustedscripturl

DOM spec:
https://github.com/whatwg/dom/pull/1268

execCommand draft:
https://w3c.github.io/editing/docs/execCommand/#methods-to-query-and-execute-commands

Blink: https://source.chromium.org/search?q=(TrustedHTML%20OR%20TrustedScript%20OR%20TrustedScriptURL%20OR%20TrustedType)%20AND%20filepath:third_party%2Fblink%2Frenderer%2Fcore%20AND%20file:idl&start=1

WebKit: https://searchfox.org/wubkat/search?q=TrustedHTML%7CTrustedScriptURL%7CTrustedScript%7CTrustedType&path=.idl%24&case=false&regexp=true

Gecko: https://searchfox.org/mozilla-central/search?q=TrustedHTML%7CTrustedScriptURL%7CTrustedScript%7CTrustedType&path=.webidl%24&case=false&regexp=true

**Comment by fred-wang** [comment]
Trusted Types spec:
* `HTMLScriptElement`'s `innerText` (`TrustedScript`)
* `HTMLScriptElement`'s `textContent` (`TrustedScript`)
* `HTMLScriptElement`'s `src` (`TrustedScriptURL`)
* `HTMLScriptElement`'s `text` (`TrustedScript`)

HTML spec:
* `Document`'s `write()` (`TrustedHTML`)
* `Document`'s `writeln()` (`TrustedHTML`)
* `Document`'s `parseHTMLUnsafe()` (`TrustedHTML`)
* `HTMLIFrameElement`'s `srcdoc` (`TrustedHTML`)
* `Document`'s `setHTMLUnsafe()̀ (`TrustedHTML`)
* `Element`'s `innerHTML` (`TrustedHTML`)
* `Element`'s `outerHTML` (`TrustedHTML`)
* `Element`'s `insertAdjacentHTML()` (`TrustedHTML`)
* `ShadowRoot`'s `setHTMLUnsafe()` (`TrustedHTML`)
* `ShadowRoot`'s `innerHTML` (`TrustedHTML`)
* `DOMParser`'s `parseFromString()̀  (`TrustedHTML`)
* `Range`'s `createContextualFragment()` (`TrustedHTML`)
* `eval()`
* `WindowOrWorkerGlobalScope`'s `setTimeout()` (`TrustedScript`)
* `WindowOrWorkerGlobalScope`'s `setInterval()` (`TrustedScript`)
* `WorkerGlobalScope`'s `importScripts()` (`TrustedScriptURL`)
* `WorkerGlobalScope`'s `importScripts()` (`TrustedScriptURL`)
* `Worker`'s constructor (`TrustedScriptURL`)
* `SharedWorker`'s constructor (`TrustedScriptURL`)

DOM spec:
* `Element`'s `setAttribute()` (`TrustedType`)
* `Element`'s `setAttributeNS()` (`TrustedType`)

CSP spec:
* `eval()` (`TrustedScript`)
* function constructor (`TrustedScript`)

execCommand draft:
* `Document`'s `execCommand()` (`TrustedHTML`)

**Comment by lukewarlow** [comment]
There's also SVGScriptElement which needs handling it's just not specced yet

**Comment by fred-wang** [comment]
@lukewarlow yes I noticed that. It seems some PRs have been merged but are still not public. will follow-up with you privately

**Comment by fred-wang** [comment]
https://github.com/w3c/trusted-types/issues/494#issuecomment-2572763334 should now be an up-to-date list of existing sinks. Still need to check whether we cover everything. The state of the SVG spec is still not great, but essentially we would make SVGScriptElement behave similarly to HTMLScriptElement.

Everything that is implemented in Gecko or WebKit is covered by one of the spec at https://github.com/w3c/trusted-types/issues/494#issuecomment-2568807108 but Chromium still seems to implement legacy stuff (probably we should write some tests to verify these are not valid sinks):

https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/dom/child_node_part.idl?q=(TrustedHTML%20OR%20TrustedScript%20OR%20TrustedScriptURL%20OR%20TrustedType)%20AND%20filepath:third_party%2Fblink%2Frenderer%2Fcore%20AND%20file:idl&start=11
https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/dom/parent_node.idl?q=(TrustedHTML%20OR%20TrustedScript%20OR%20TrustedScriptURL%20OR%20TrustedType)%20AND%20filepath:third_party%2Fblink%2Frenderer%2Fcore%20AND%20file:idl&start=11
https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/core/dom/child_node.idl?q=(TrustedHTML%20OR%20TrustedScript%20OR%20TrustedScriptURL%20OR%20TrustedType)%20AND%20filepath:third_party%2Fblink%2Frenderer%2Fcore%20AND%20file:idl&start=11

**Comment by fred-wang** [comment]
> Everything that is implemented in Gecko or WebKit is covered by one of the spec at https://github.com/w3c/trusted-types/issues/494#issuecomment-2568807108 but Chromium still seems to implement legacy stuff (probably we should write some tests to verify these are not valid sinks).

Tests added in https://github.com/web-platform-tests/wpt/pull/49920 for ParentNode/ChildNode and support remove from Chromium for DOM parts in https://chromium-review.googlesource.com/c/chromium/src/+/6150068.

**Comment by fred-wang** [comment]
I'm closing this issue as complete. Potential improvements that can be done separately:
* Refine tests about content handler attributes for example https://github.com/w3c/trusted-types/issues/573
* Refine tests for SVG attributes, once the specs have more details on script enforcement.
* Do similar tests for violation object samples.
